internal/app/service.go

```diff
@@ -430,6 +430,22 @@ func (svc *Service) AddFunctionsPipelineForTopics(id string, topics []string, tr
 	return nil
 }
 
+// SetFunctionsPipelineTransforms updates the functions pipeline for the specified id
+func (svc *Service) SetFunctionsPipelineTransforms(id string, transforms ...interfaces.AppFunction) error {
+	if len(transforms) == 0 {
+		return errors.New("no transforms provided to pipeline")
+	}
+
+	svc.runtime.SetFunctionsPipelineTransforms(id, transforms)
+	svc.lc.Debugf("Pipeline '%s' updated with %d transform(s)", id, len(transforms))
+	return nil
+}
+
+// ClearAllFunctionsPipelineTransforms clears all the functions pipelines
+func (svc *Service) ClearAllFunctionsPipelineTransforms() {
```

Comment:
ClearAllFunctionsPipelineTransforms` leaves the pipelines in place without any transforms. 

What we really need is `RemoveAllFunctionPipelines` which calls a new function on the runtime to remove all previously added pipelines. `